python/ray/air/_internal/tensorflow_utils.py

```diff
@@ -65,8 +65,20 @@ def convert_pandas_to_tf_tensor(
     def tensorize(series):
         try:
             return tf.convert_to_tensor(series, dtype=dtype)
-        except ValueError:
-            return tf.stack([tensorize(element) for element in series])
+        except ValueError as outer_e:
+            # This exception will be raised if series is of object dtype or otherwise
+            # cannot be made into a tensor directly. We assume it's a sequence in that
+            # case. This is more robust than checking for dtype.
+            tensors = [tensorize(element) for element in series]
+            try:
+                return tf.stack(tensors)
+            except Exception:
```

Comment:
i think both `tf.stack` and `tf.ragged.stack` raises `ValueError`:

https://www.tensorflow.org/api_docs/python/tf/stack
https://www.tensorflow.org/api_docs/python/tf/ragged/stack

Reply:
Hmm I thought that I saw other exceptions leak out, but I'll try reducing the exception scope to `ValueError` and see if it suffices!

Reply:
In any case, we want this to be a blanket errors catch-all anyways, where we want to:
1. try to convert, and if that fails ->
2. try to convert individual elements and stack, and if that fails ->
3. try to construct a ragged tensor, and if that fails ->
4. raise an error capturing our attempted conversions and their respective failures.

If (2) and (3) don't have a well-defined set of exception types for failures modes on data we expect to see, I think that a blanket `except Exception` catch-all is fine.

Reply:
Ok, I think we can start with `except Exception`.